api/http_lifecycle_test.go

```diff
@@ -11,23 +11,20 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-type NopEndpointSource struct {
-	Root string
-}
-
-func (e *NopEndpointSource) Endpoint(op string) config.Endpoint {
-	return config.Endpoint{Url: e.Root}
-}
-
 var (
-	source = &NopEndpointSource{"https://example.com"}
+	testConfig = NewTestConfig()
 )
 
+func NewTestConfig() *config.Configuration {
+	c := config.NewFrom(config.Values{})
+	c.SetManualEndpoint(config.Endpoint{Url: "https://example.com"})
+	return c
+}
 func TestHttpLifecycleMakesRequestsAgainstAbsolutePath(t *testing.T) {
 	SetupTestCredentialsFunc()
 	defer RestoreCredentialsFunc()
 
-	l := api.NewHttpLifecycle(source)
+	l := api.NewHttpLifecycle(testConfig)
```

Comment:
Continuing my 🚋  of 💭  from the above comment, I'd be 👍  with keeping the `NewTestConfig()` function and just inlining a call to it here:

```go
l := api.NewHttpLifecycle(NewTestConfig())
```